conda/core/package_cache.py

```diff
@@ -376,7 +377,7 @@ def make_actions_for_dist(dist, record):
         first_writable_cache = PackageCache.first_writable()
         pc_entry_writable_cache = first(
             (writable_cache.get(dist) for writable_cache in PackageCache.all_writable()),
-            key=lambda pce: pce and pce.is_fetched and pce.tarball_matches_md5(record.md5)
+            key=lambda d: d and d.is_fetched and d.tarball_matches_md5_if(md5)
```

Comment:
These should still be package cache entries, not dists, I think.  Let's keep the lambda argument `pce` rather than `d`.